docs/README.md

```diff
@@ -1,14 +1,411 @@
+
+```tsx
+import {
+	PaymentProvider,
+	// ...
+} from '@vtex/payment-provider'
+
+class YourPaymentConnector extends PaymentProvider {
+
+	// ... implementation of the other routes functions
+}
+```
+
+Typescript should automatically check for typing errors, but if you need, you can check the requests and responses signatures [here](https://developers.vtex.com/vtex-developer-docs/reference/payment-flow).
+
+
+<br />
+
+### Payment Provider Builder
+
+In order to specify which payment methods the connector will accept to process, you need to create a folder named *paymentProvider* using the following folder structure
+
+```markdown
+node
+paymentProvider
+manifest.json
+```
+
+Then, inside paymentProvider folder you must create a file named *configuration.json*
+
+```markdown
+node
+paymentProvider
+   |--configuration.json
+manifest.json
+```
+
+Next, declare the accepted payment methods, for instance:
+
+```json
+{
+  "name": "MyConnector",
+  "paymentMethods": [
+    {
+      "name": "Visa",
+      "allowsSplit": "onCapture"
+    },
+    {
+      "name": "American Express",
+      "allowsSplit": "onCapture"
+    },
+    {
+      "name": "Diners",
+      "allowsSplit": "onCapture"
+    },
+    {
+      "name": "Elo",
+      "allowsSplit": "onCapture"
+    },
+    {
+      "name": "Hipercard",
+      "allowsSplit": "onCapture"
+    },
+    {
+      "name": "Mastercard",
+      "allowsSplit": "onCapture"
+    },
+    {
+      "name": "BankInvoice",
+      "allowsSplit": "onAuthorize"
+    }
+  ]
+}
+```
+
+By doing this you don't need to declare /manifest or /payment-methods route, it will be implemented automatically by the builder.
+
+Detailed flow [here](https://miro.com/app/board/o9J_kxgli2c=/?moveToWidget=3074457355130162795&cot=14)
+
+![building-a-payment-provider](https://user-images.githubusercontent.com/27698855/116406939-9603c900-a807-11eb-857b-babe02967f95.png)
+
+<br />
+
+### Overriding Default Routes
+
+You can also override the default generated routes by redeclaring it on *service.json*
+
+```json
+{
+  "memory": 256,
+  "ttl": 10,
+  "timeout": 10,
+  "minReplicas": 2,
+  "maxReplicas": 3,
+  "routes": {
+    "authorize": {
+      "path": "/**_v/api/my-connector/**payments",
+      "public": true
+    },
+    "cancel": {
+      "path": "/**_v/api/my-connector/**payments/:paymentId/cancellations",
+      "public": true
+    },
+    "settle": {
+      "path": "/**_v/api/my-connector/**payments/:paymentId/settlements",
+      "public": true
+    },
+    "refund": {
+      "path": "/**_v/api/my-connector/**payments/:paymentId/refunds",
+      "public": true
+    },
+    "inbound": {
+      "path": "/**_v/api/my-connector/**payments/:paymentId/inbound/hooks",
+      "public": true
+    },
+  }
+}
+```
+
+Then you must specify which is the new serviceUrl on paymentProvider/configuration.json
+
+```json
+{
+  "name": "MyConnector",
+  **"serviceUrl": "/_v/api/my-connector",**
```

Comment:
I think this is not working on markdown 
![image](https://user-images.githubusercontent.com/5839364/116453742-9ca93500-a835-11eb-84f3-0a8290b778a6.png)

Its fine to remove it if theres no way to do it